core/src/util/extract_option.rs

```diff
@@ -0,0 +1,209 @@
+//! Extract type `T` from an `Option<T>`
+//!
+//! Procedural macros often have to hard-code support for `Option`s, so much so that the logic
+//! has been re-implemented [hunreds](https://github.com/search?q=path%3A*.rs+%2Ffn+.*opt.*%5C%28.*%3A+.*syn%3A%3AType%5C%29+-%3E+Option%3C.*syn%3A%3AType%3E%2F&type=code) of times across the ecosystem.
+//!
+//! Considering this common need, darling provides 3 functions for extracting a [`Type`] from an `Option<T>`:
+//!
+//! - [`from_owned`]: `fn(Type) -> Result<Type>`
+//! - [`from_mut`]: `fn(&mut Type) -> Result<&mut Type>`
+//! - [`from_ref`]: `fn(&Type) -> Result<&Type>`
```

Comment:
```suggestion
```

The justification is great for the GitHub issue, but unnecessary in the module documentation.

Listing the different signatures in the module-level docs is unnecessary, since on docs.rs the functions will be listed directly below.